sklearn/tree/_splitter.pyx

```diff
@@ -335,6 +335,8 @@ cdef inline int node_split_best(
         features[f_i], features[f_j] = features[f_j], features[f_i]
 
         # Evaluate all splits
+        # The criterion has a view into the samples that was sorted by the partitioner.
+        # The criterion will use that ordering when evaluting the splits.
```

Comment:
```suggestion
        # The criterion will use that ordering when evaluating the splits.
```

Reply:
Nit: Would it be more clear that this is just related to just this point of the code and not `criterion.reset()`? E.g. 

`# At this point, the criterion object has a view into the samples that was sorted by the partitioner.`

Feel free to veto me.